fvm/src/machine/engine.rs

```diff
@@ -71,9 +81,32 @@ impl Default for MultiEngine {
     }
 }
 
-pub fn default_wasmtime_config() -> wasmtime::Config {
+pub fn default_wasmtime_config(
+    instance_count: u32,
+    instance_memory_maximum_size: u64,
```

Comment:
These should ideally be defaults that we adjust later.

Reply:
I did it this way for a couple of reasons:
1. The defaults are defined on the `NetworkConfig` and it seemed better to have one set of defaults than two that can get out of sync and cause confusion.
2. As you see the `instance_memory_maximum_size` is used in multiple places: inside `InstanceLimits` and to set the value of `static_memory_maximum_size`. It seems like it would be difficult to ask that everyone using the config and wanting to set this remembers to set both values correctly. 
3. The parameters passed to `allocation_strategy` are more complex and the `instance_count` is just one of them; what we actually want to set is `EngineConfig::max_call_depth`, but this again is a dependent setting that you have to get right with all the other unrelated defaults like the affinity, and you can't just set it as a field, as it's part of an enum.

It seemed to me like the cleanest way to keep all relevant configuration visible in this method is to ask for these two values to be given to the method.

This method is only ever called from `Engine::new_default`, so maybe we could just make it private to this module, and then the scope of "later" is very clear.

Reply:
Yeah, makes sense. See https://github.com/filecoin-project/ref-fvm/pull/1073.